src/Type/Php/MinMaxFunctionReturnTypeExtension.php

```diff
@@ -111,18 +111,13 @@ private function processArrayType(string $functionName, Type $argType): Type
 					$resultTypes[] = new ConstantBooleanType(false);
 					continue;
 				}
-				$iterableValueType = $constArrayType->getIterableValueType();
 				$argumentTypes = [];
 				if (!$isIterable->yes()) {
 					$argumentTypes[] = new ConstantBooleanType(false);
 				}
 
-				if ($iterableValueType instanceof UnionType) {
-					foreach ($iterableValueType->getTypes() as $innerType) {
-						$argumentTypes[] = $innerType;
-					}
-				} else {
-					$argumentTypes[] = $iterableValueType;
+				foreach ($constArrayType->getValueTypes() as $innerType) {
```

Comment:
Finally! This is what I meant that you don't need UnionType here anymore. I'm happy to merge this if the build is green.